### PR TITLE
Bug/remove notifications singleton

### PR DESCRIPTION
**Pull request by myieye**
Waiting on #363 

This presumably isn't currently causing trouble, because we probably only call our notification api client-side, but we don't want it to be possible for notifications to leak between clients.

### CODE REVIEW

File: frontend/src/lib/notify/index.ts
Comment (hahn-kev) [inline frontend/src/lib/notify/index.ts:1]: it's not my favorite that the class functions here conflict with the function style we use everywhere else, but it looks like it's unavoidable without rewriting a lot of other code

File: frontend/src/routes/+layout.svelte [hunk 1]
```diff
@@ -16,6 +16,8 @@
   export let data: LayoutData;
   const { page, updated } = getStores();
 
+  const { notifyWarning } = initNotificationService(new NotificationService(writable([])));
```
Comment (hahn-kev) [inline frontend/src/routes/+layout.svelte:19]: it feels weird for this to be where the value of the notification service is declared, specifically what's passed into the init, the rest is fine.

**Review by hahn-kev — APPROVED** [review]

File: frontend/src/routes/+layout.svelte (hunk 1 above)
Comment (myieye) [reply frontend/src/routes/+layout.svelte]: True. There's no reason why we can't specify the context value when we define the context.
I refactored it and added 57e68c7.

File: frontend/src/lib/notify/index.ts
Comment (myieye) [reply frontend/src/lib/notify/index.ts:1]: 🙈

**Pull request merged**